src/internal/clusterstate/v2.12.0/snapshot.go

```diff
@@ -15,7 +15,7 @@ func createSnapshotSchema(ctx context.Context, env migrations.Env) error {
 	if _, err := tx.ExecContext(ctx, `create table recovery.snapshots (
 		id bigserial not null primary key,
 		chunkset_id bigint not null references storage.chunksets(id),
-		sql_dump_fileset_id uuid null references storage.filesets(id),
+		sql_dump_fileset_pin_id bigint null references storage.fileset_pins(id),
```

Comment:
I don't think we should be creating foreign key constraints on database tables in different subsystems. The database tables are an implementation detail of the subsystem. The subsystem API should be the only way to interface with a subsystem to ensure we can make changes to the subsystem implementation without having to migrate a bunch of dependent subsystems. Also, you could probably just shorten the names to pin rather than pin_id. The external concept should just be the integer, which is the pin.

Reply:
Shortened the names and removed the foreign keys.

I think we're misleading ourselves if we think pachyderm has isolated subsystems.  It's a monolith and would look very different to end users if we ever completed this work.  Maybe in pachyderm 3.0 ;)  In the meantime, I think that removing enforced referential integrity hides bugs.